### PR TITLE
New version plugin aa

### PR DESCRIPTION
**Pull request by ArtemAnoshin**

### CODE REVIEW

File: Classes/Hooks/Form.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+use TYPO3\CMS\Form\Domain\Runtime\FormRuntime;
+
+class Form
+{
+    public function afterSubmit(FormRuntime $runtime, $element, $value, $requestArguments)
+    {
+        global $cleantalk_execute;
+
+        if ($this->needProcess($runtime) && !$cleantalk_execute) {
+            $helper = new Helper();
```
Comment (Glomberg) [inline Classes/Hooks/Form.php]: Not need to instantiate the Helper object.

File: ext_emconf.php [hunk 2]
```diff
@@ -1,36 +1,28 @@
...
-		),
-		'conflicts' => array(
-		),
-		'suggests' => array(
-		),
-	),
-);
\ No newline at end of file
+$EM_CONF['cleantalk'] = [
+    'title' => 'CleanTalk antispam',
```
Comment (Glomberg) [inline ext_emconf.php]: Not correspond by our brand book. Need to change the terms: anti-spam

File: ext_emconf.php [hunk 3]
```diff
@@ -1,36 +1,28 @@
...
+    'description' => 'Protect your site from spam',
+    'category' => 'module',
+    'author' => 'Cleantalk',
```
Comment (Glomberg) [inline ext_emconf.php]: Need to change the terms: CleanTalk

File: ext_emconf.php [hunk 4]
```diff
@@ -1,36 +1,28 @@
...
+    'author_email' => 'welcome@cleantalk.org',
+    'author_company' => 'Cleantalk',
```
Comment (Glomberg) [inline ext_emconf.php]: Need to change the terms: CleanTalk

File: ext_tables.php [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?php
-if (!defined('TYPO3_MODE')) {
-	die('Access denied.');
-}
-
-\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addStaticFile($_EXTKEY, 'Configuration/TypoScript', 'CleanTalk antispam');
+//if (!defined('TYPO3_MODE')) {
```
Comment (Glomberg) [inline ext_tables.php]: Please remove this 'died code'

File: ext_tables.sql [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
-  PRIMARY KEY (`server_id`)
-) ENGINE=MyISAM DEFAULT CHARSET=utf8;
-
-DROP TABLE IF EXISTS `cleantalk_timelabels`;
-CREATE TABLE `cleantalk_timelabels` (
-  `ct_key` varchar(255) NOT NULL default 'mail_error',
-  `ct_value` int(11),
-  PRIMARY KEY (`ct_key`)
-) ENGINE=MyISAM DEFAULT CHARSET=utf8;
+# DROP TABLE IF EXISTS `cleantalk_server`;
```
Comment (Glomberg) [inline ext_tables.sql]: Please remove this 'died code'

File: lib/Cleantalk/Custom/Antispam/Cleantalk.php [hunk 7]
```diff
@@ -0,0 +1,859 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Antispam/Cleantalk.php:1]: Why this class is in `Custom` namespace? Is not it really enough the common class?

File: lib/Cleantalk/Custom/Antispam/CleantalkRequest.php [hunk 8]
```diff
@@ -0,0 +1,202 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Antispam/CleantalkRequest.php]: Why this class is in `Custom` namespace? Is not it really enough the common class?

File: lib/Cleantalk/Custom/Antispam/CleantalkResponse.php [hunk 9]
```diff
@@ -0,0 +1,181 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Antispam/CleantalkResponse.php]: Why this class is in `Custom` namespace? Is not it really enough the common class?

File: lib/Cleantalk/Custom/Antispam/lock-page-ct-die.html [hunk 10]
```diff
@@ -0,0 +1,150 @@
+<!DOCTYPE html>
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Antispam/lock-page-ct-die.html:1]: Why this class is in `Custom` namespace? Is not it really enough the common class?

File: lib/Cleantalk/Custom/Cron/Cron.php [hunk 11]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Cron/Cron.php]: It seems Cron functionality not working - Drupal handlers used. Need to be recheck and fix

File: lib/Cleantalk/Custom/Db/Db.php [hunk 12]
```diff
@@ -0,0 +1,95 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Db/Db.php]: This class uses Drupal code. Fix it.

File: lib/Cleantalk/Custom/Firewall/Modules/AntiCrawler.php [hunk 13]
```diff
@@ -0,0 +1,326 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Firewall/Modules/AntiCrawler.php]: The AC need to be removed

File: lib/Cleantalk/Custom/Firewall/Modules/AntiFlood.php [hunk 14]
```diff
@@ -0,0 +1,272 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/Firewall/Modules/AntiFlood.php]: The AF need to be removed

File: lib/Cleantalk/Custom/RemoteCalls/RemoteCalls.php [hunk 15]
```diff
@@ -0,0 +1,74 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/RemoteCalls/RemoteCalls.php]: Drupal code here

File: lib/Cleantalk/Custom/StorageHandler/StorageHandler.php [hunk 16]
```diff
@@ -0,0 +1,36 @@
+<?php
```
Comment (Glomberg) [inline lib/Cleantalk/Custom/StorageHandler/StorageHandler.php]: Druapl code here

**Review by Glomberg — CHANGES_REQUESTED** [review]
Need major fixes

File: Classes/Hooks/Form.php (hunk 1 above)
Comment (ArtemAnoshin) [reply Classes/Hooks/Form.php]: Resolved

File: ext_emconf.php (hunk 2 above)
Comment (ArtemAnoshin) [reply ext_emconf.php]: Resolved

File: ext_emconf.php (hunk 3 above)
Comment (ArtemAnoshin) [reply ext_emconf.php]: Resolved

File: ext_emconf.php (hunk 4 above)
Comment (ArtemAnoshin) [reply ext_emconf.php]: Resolved

File: ext_tables.php (hunk 5 above)
Comment (ArtemAnoshin) [reply ext_tables.php]: Resolved

File: ext_tables.sql (hunk 6 above)
Comment (ArtemAnoshin) [reply ext_tables.sql]: Resolved

File: lib/Cleantalk/Custom/Antispam/CleantalkRequest.php (hunk 8 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Custom/Antispam/CleantalkRequest.php]: I'll try to solve

File: lib/Cleantalk/Custom/Antispam/CleantalkResponse.php (hunk 9 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Custom/Antispam/CleantalkResponse.php]: I'll try to solve

File: lib/Cleantalk/Custom/Antispam/lock-page-ct-die.html (hunk 10 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Custom/Antispam/lock-page-ct-die.html:1]: relative path

File: lib/Cleantalk/Custom/Cron/Cron.php (hunk 11 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Custom/Cron/Cron.php]: Resolved

File: lib/Cleantalk/Custom/Db/Db.php (hunk 12 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Custom/Db/Db.php]: Resolved

File: composer.json [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+  "type": "typo3-cms-extension",
+  "description": "Protect your site from spam",
+  "license": "GPL-2.0-or-later",
+  "version": "1.0.0",
+  "require": {
+    "typo3/cms-core": "^10.4 || ^11.0"
+  },
+  "require-dev": {
+    "cleantalk/antispam": "*",
+    "cleantalk/firewall": "*",
```
Comment (Glomberg) [inline composer.json]: `cleantalk/firewall` and `cleantalk/storage-handler` not necessary on this project. Remove these requires.

**Review by Glomberg — CHANGES_REQUESTED** [review]
There are too much extra and unnecessary files in the lib directory. Please, rebuild the lib directory and make it more cleaner.

**Review by Glomberg — APPROVED** [review]

**Pull request merged**